UKHomeOffice/nmsw-ui
# NMSW-271 Upload crew details (FAL 5)

# Ticket

NMSW-271

----

## AC

Given I have uploaded a file
And There are no errors
When I click check for errors
Then I am shown the no errors found page

Given a user has not uploaded a file
When they click check for errors
Then a form field styled error is shown

Given a user has uploaded a file not of type .csv or .xlsx
When they click check for errors
Then a form field styled error is shown

Given a user has uploaded a file with a size greater than 4MB
When they click check for errors
Then a form field styled error is shown

----

## To test

- Log in
- Click on `Report a voyage` button
- Upload a general declaration FAL 1
- Go through the process until you see the task list
- Click on "Crew details including supernumeraries (FAL 5)" link
- You should see the Upload the Crew details page that matches the prototype
- Click on check for errors
- Error of “Select a Crew details including supernumeraries (FAL 5)” should be shown
- Upload an image file
- Click on `Check for errors`
- Error of “The selected file must be a csv or xlsx” should be shown
- Upload a large file (over 4MB)
- Click on `Check for errors`
- Error of “The file must be smaller than 4MB” should be shown
- Click on `The file must be smaller than 4MB` link in the `Error summary panel`
- You should be scrolled to the Upload a file question
- Upload a new file
- The error message should clear
- Delete your auth token
- Click on `Check for errors`
- You should be redirected to sign in
- Sign in
- You should directed back to the Upload Crew details including supernumeraries (FAL 5) page

For dev testing 404
- In `FileUploadForm` change the endpoint to `/declaration//upload-fal5`
- Upload a file
- Click `Check for errors`
- You should be taken to something went wrong page

SUCCESS SCENARIO:
- Upload the Crew details template file
- Click `Check for errors`
- You should be taken to the confirmation of upload page with a success message
- Click `Save and continue`
- You should be taken to the task details page

----

## Notes
